[DEVTOOLS-230] Migrate from TSLint to ESLint

kmonahan opened the pull request:
Replaces TSLint with ESLint in the generator itself and in the JavaScript projects created. There were two places where I found some relevant differences between TSLint and ESLint:

1. ESLint's prefer template strings rule does not have the option to allow single concatenations that TSLint did. I've updated the code so that it meets the stricter standards of ESLint. It's a little bit annoying, but I think I actually like it better: it's a lot easier to tell whether or not there's a space between concatenated bits when you use a template string.

2. ESLint's sort-imports rule doesn't understand groups of imports the way TSLint does. As a result, I disabled the rule to sort members, though kept the one to require `import {a, b, c} from 'foo'} versus `import {b, a, c} from 'foo'`. Open to other POVs on this as I decided this based on personal preference--I like being able to organize imports into related groups if I have a bunch and don't particularly care if they're alphabetized within that group.

gustavderdrache commented on .eslintrc.js:
```diff
@@ -0,0 +1,75 @@
+        "constructor-super": "error",
+        "curly": "error",
+        "dot-notation": "error",
+        "eol-last": "off",
```
Is there a reason these are disabled in the configuration instead of using `eslint-config-prettier`? It looks like you'd only need to add the plugin to your extends array.

kmonahan replied:
Ah, I see what you're saying. No, no special reason other than "that's what npx tslint-to-eslint-config gave me." I'm good with using eslint-config-prettier and shortening this list a bit.

gustavderdrache replied:
Let's do that (and for the forumone-cli PR as well).

kmonahan commented:
A couple updates here:
1. Took out the items that were redundant with the eslint-config-prettier and added the typescript-specific prettier config to the extends list.
2. Extended eslint and @typescript-eslint's recommended configs to cover basic rules and removed rules that were then redundant. Goal here is to avoid having to specify all best practices in the eslint config file itself.
3. Turned off a couple rules in the base `.eslintrc` file that were newly added as part of the base configs and conflicted with the existing code. In the future, we might consider making the linting a bit stricter and doing some refactoring, but my goal for now is just to enforce code consistency and some good habits going forward.

I still need to go back and do the same to the forumone-cli PR.
